polkadot/runtime/common/src/crowdloan/mod.rs

```diff
@@ -576,7 +577,10 @@ pub mod pallet {
 			// can take care of that.
 			debug_assert!(Self::contribution_iterator(fund.fund_index).count().is_zero());
 
-			frame_system::Pallet::<T>::dec_providers(&Self::fund_account_id(fund.fund_index))?;
+			// Crowdloan over, burn all funds.
+			let _imba = CurrencyOf::<T>::make_free_balance_be(&pot, Zero::zero());
```

Comment:
dq but I don't get why this is needed; a CL is dissolve-able only when all funds are already returned, no? if not, I think it should be like this.

seems to be checked as a part of `let can_dissolve = permitted && fund.raised.is_zero();`

So the invariant is `fund.raised` should always be equal to `CurrencyOf::<T>::free_balance(&pot)`?

Reply:
Ahh okay, reading the test I realize that this is catering for the cases where a manual transfer happens to the fund account.